monitor.go

```diff
@@ -0,0 +1,100 @@
+package starter
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+func ExampleMonitor() {
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	ch := make(chan *exec.Cmd, 10)
+	done := make(chan *exec.Cmd)
+	go monitor(ctx, ch, done)
+
+	for i := 0; i < 10; i++ {
+		cmd := exec.Command("sleep", strconv.Itoa(i))
+		cmd.Start()
+		ch <- cmd
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			fmt.Printf("timeout reached\n")
+			return
+		case cmd, ok := <-done:
+			if !ok {
+				fmt.Println("monitor exited")
+				return
+			}
+			fmt.Printf("notified: %d\n", cmd.ProcessState.Pid())
+		}
+	}
+}
+
+// monitor is a process (i.e. *exec.Cmd) monitor. it asynchronously
+// listens for either (a) one of the monitored process exits, or (b)
+// we get a request to watch for a new worker.
+//
+// users of this function can "wait" for the exit of a command
+// by checking the done channel
+func monitor(ctx context.Context, src chan *exec.Cmd, done chan *exec.Cmd) {
+	defer close(done)
```

Comment:
Yes we do. https://github.com/lestrrat/go-server-starter/pull/11/commits/bb58920e7ea6f6c311dcb082c0814fa5c01a5a61